### PR TITLE
Added gem-boosted powers to move description

### PR DESCRIPTION
**Pull request by hvk3**
A small change to ui

### CODE REVIEW

**Comment by panpawn** [comment]
Next time, try to rebase instead of closing / opening a new pull request

**Comment by Zarel** [comment]
I even asked you not to create a new pull request when we talked about it the other day...

**Comment by hvk3** [comment]
Added support for all items

**Comment by Zarel** [comment]
@Joimer, thoughts?

I think we might want this to be its own file, if it's getting to be this complicated.

**Comment by Joimer** [comment]
Yeah, it's getting pretty big and that makes the file less readable, and it's already pretty big. How about a file for tooltip information processing?

**Comment by Zarel** [comment]
Full support

**Comment by hvk3** [comment]
What should be done next?

**Comment by Marty-D** [comment]
Couldn't you just move [this block of code](https://github.com/Zarel/Pokemon-Showdown-Client/pull/574/files#diff-bca2828a0ba7901d7d7c95b095afd014R1335) to below [this line](https://github.com/Zarel/Pokemon-Showdown-Client/pull/574/files#diff-bca2828a0ba7901d7d7c95b095afd014R1913), and remove all the duplication in between? Am I missing something? This could also probably be done more simply by including the check for `basePower` in the first check instead of everywhere else, using switch statements, etc.

**Comment by hvk3** [comment]
The point is that some of the moves don't have base powers initially, and were added as an exception before I started. So I had to duplicate some of the stuff.

**Comment by Marty-D** [comment]
Does that matter? They will all have `basePower` defined by the time it gets to the return statement, which is where this should be done so duplication becomes unnecessary.

**Comment by hvk3** [comment]
Okay, so moves with basePower can be handled as you said. There'll be some duplication for moves with ranges then, I guess.

**Comment by hvk3** [comment]
Removed redundancy from code

**Comment by hvk3** [comment]
Corrected electroball range issue as well

**Comment by hvk3** [comment]
Moved code to client-battle-tooltips.js

**Comment by hvk3** [comment]
Modifications according to Quite Quiet and E4 Flint

**Comment by Zarel** [comment]
Can you give an MIT license? Post:

> I, __, hereby license all my contributions to the Pokémon Showdown project, past and future, under the MIT license.

**Comment by hvk3** [comment]
I, Harshvardhan Kalra, hereby license all my contributions to the Pokémon Showdown project, past and future, under the MIT license.

File: js/client-battle.js [hunk 1]
```diff
@@ -234,7 +234,7 @@
 
 			if (this.battle.ended) {
 
-				var replayDownloadButton = '<a href="//replay.pokemonshowdown.com/" class="button replayDownloadButton" style="float:right;padding:2px 6px"><i class="fa fa-download"></i> Download replay</a>';
+				var replayDownloadButton = '<span style="float:right;"><a href="//replay.pokemonshowdown.com/" class="button replayDownloadButton" style="padding:2px 6px"><i class="fa fa-download"></i> Download replay</a><br /><br /><button name="saveReplay"><i class="fa fa-upload"></i> Upload and share replay</button></span>';
```
Comment (Zarel) [inline js/client-battle.js]: What's going on here? Why was this changed?

Comment (Zarel) [inline js/client-battle.js]: Well, at least it merges cleanly, but ideally you should rebase on latest master for a cleaner blame history.

**Comment by hvk3** [comment]
Some commit that i had not merged possibly

**Comment by hvk3** [comment]
Small corrections

**Comment by Slayer95** [comment]
@Zarel, the new file wasn't added to the index.
@hvk3 should be sending a follow-up PR soon, but this is a heads-up so that the client isn't left in a bad state after pulling the changes.

**Comment by Slayer95** [comment]
lol I'm slow. Merged.
